src/main/java/com/google/sps/data/AvailabilityTimeSlotGenerator.java

```diff
@@ -0,0 +1,158 @@
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.TextStyle;
+import java.util.Locale;
+import java.time.format.DateTimeFormatter;
+
+// TODO: Add a test for the constructor.
+
+/** A generator of a collection of AvailabilityTimeSlot Objects. */
+public class AvailabilityTimeSlotGenerator {
+  private static final int numberOfSlotsPerDay = 48;
+  private static final int TIME_8AM = 8;
+  private static final int TIME_7PM = 19;
+
+  /**
+   * Constructs a list of a day's worth of AvailabilityTimeSlot objects.
+   *
+   * @param instant An instant on the day that time slots are generate for.
+   * @param timezoneOffset A String that represents the difference between UTC and the user's
+   *     current timezone. Example: A user in EST has a timezoneOffset of "-240" which means that
+   *     EST is 240 minutes behind UTC.
+   */
+  // TODO: Create a timeSlotsForWeek method.
+  public static List<AvailabilityTimeSlot> timeSlotsForDay(Instant instant, String timezoneOffset) {
+    ZonedDateTime day = generateDay(instant, timezoneOffset);
+    ZoneId zoneId = day.getZone();
+    String dayOfWeek = day.getDayOfWeek().getDisplayName(TextStyle.SHORT, Locale.US);
+    int year = day.getYear();
+    int month = day.getMonthValue();
+    int dayOfMonth = day.getDayOfMonth();
+
+    String date = generateDate(dayOfWeek, month, dayOfMonth);
+    List<String> times = generateTimes();
+    List<String> utcEncodings = generateUTCEncodings(year, month, dayOfMonth, zoneId);
+    List<Boolean> selectedStatuses = getSelectedStatuses();
+    return generateTimeSlots(utcEncodings, times, date, selectedStatuses);
+  }
+
+  // Uses an Instant and a timezoneOffset String to create a ZonedDateTime instance
+  // for the day specified by the Instant.
+  private static ZonedDateTime generateDay(Instant instant, String timezoneOffset) {
+    ZoneOffset timeZoneOffset = convertStringToOffset(timezoneOffset);
+    ZoneId zoneId = ZoneId.ofOffset("UTC", timeZoneOffset);
+    return instant.atZone(zoneId);
+  }
+
+  // This method takes the timezoneOffset String and converts it
+  // into a proper ZoneOffset instance.
+  private static ZoneOffset convertStringToOffset(String timezoneOffset) {
```

Comment:
How do you handle unparsable inputs here? Consider tests?

(This is made obsolete by a comment on timeSlotsForDay taking an integer offsetMinutes instead of a string)

Reply:
Done